File: lab5/C++/src/Vehicle.cpp
#include "Vehicle.hpp"
#include "SupplyRequestDetail.hpp"

Vehicle::Vehicle(int id, const std::string& name, VehicleType type, FuelType fuelType, int count)
    : Equipment(id, name, count), m_type(type), m_fuelType(fuelType) {
}


bool Vehicle::IsMatches(SupplyRequestDetail* detail) const {
    return detail->GetSupplyType() == SupplyType::Vehicle &&
        detail->GetVehicleType() == m_type &&
        detail->GetFuelType() == m_fuelType;
}

VehicleType Vehicle::GetType() const {
    return m_type;
}

FuelType Vehicle::GetFuelType() const {
    return m_fuelType;
}

File: lab6/Java/src/warehouse/equipments/Vehicle.java
package warehouse.equipments;

import enums.FuelType;
import enums.SupplyType;
import enums.VehicleType;
import java.util.Random;
import request.SupplyRequestDetail;

public class Vehicle extends Equipment implements Cloneable {
    VehicleType type;
    FuelType fuelType;

    public Vehicle(int id, String name, VehicleType type, FuelType fuelType, int count)
    {
        super(id, name, count);
        this.type = type;
        this.fuelType = fuelType;
    }

    @Override
    public Boolean isMatches(SupplyRequestDetail detail) {
        return detail.getSupplyType() == SupplyType.VEHICLE &&
               detail.getVehicleType() == type &&
               detail.getFuelType() == fuelType;
    }

    public  VehicleType getType() {
        return type;
    }

    public  FuelType getFuelType() {
        return fuelType;
    }

    @Override
    public Vehicle clone() throws CloneNotSupportedException{
        var clone = (Vehicle)super.clone();
        clone.id = new Random().nextInt(0,9999);
        return clone;
    }
}
